components/worktemplates/components/menu.tsx

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useEffect, useState} from 'react';
+import {useDispatch, useSelector} from 'react-redux';
+import styled from 'styled-components';
+
+import {useIntl} from 'react-intl';
+
+import classNames from 'classnames';
+
+import {Category, WorkTemplate} from '@mattermost/types/worktemplates';
+import {worktemplates as workTemplateDevData} from '../dev_data';
+
+import {injectDevData} from 'mattermost-redux/actions/worktemplates';
+import {GlobalState} from '@mattermost/types/store';
+
+import UseCaseMenuItem from './menu/use_case';
+
+const Categories = styled.div`
+    h2 {
+        margin: 0;
+        padding: 8px 16px;
+
+        font-weight: 600;
+        font-size: 12px;
+        line-height: 16px;
+        letter-spacing: 0.02em;
+        text-transform: uppercase;
+    }
+
+    ul {
+        list-style: none;
+        padding: 0;
+        width: 176px;
+    }
+`;
+
+const CategoryButton = styled.button`
+    width: 155px;
+    padding: 10px 16px;
+    border: 0;
+    background: none;
+    text-align: left;
+
+    &.selected {
+        background: rgba(var(--denim-button-bg-rgb), 0.04);
+        font-weight: 600;
+        color: var(--denim-button-bg);
+        cursor: pointer;
+    }
+`;
+
+const UseCases = styled.div`
+    display: flex;
+    justify-content: space-between;
+    flex-wrap: wrap;
+    width: 692px;
+`;
+
+interface MenuProps {
+    className?: string;
+    onTemplateSelected: (template: WorkTemplate, quickUse: boolean) => void;
+}
+
+const Menu = (props: MenuProps) => {
+    const {formatMessage} = useIntl();
+
+    const [selectedCategory, setSelectedCategory] = useState('');
+
+    const dispatch = useDispatch();
+    const categories = useSelector((state: GlobalState) => state.entities.worktemplates.categories);
+    const worktemplates = useSelector((state: GlobalState) => state.entities.worktemplates.templatesInCategory);
+
+    useEffect(() => {
+        // DEV - REMOVE ME WHEN SERVER RETURNS DATA
+        // allows us to load the dev work templates from the store
+        dispatch(injectDevData(workTemplateDevData));
+    }, []);
+
+    useEffect(() => {
+        if (categories.length === 0 || selectedCategory !== '') {
```

Comment:
useSelector can potentially return undefined, for defensive programming I would set this as `categories?.length`